src/main/java/org/eclipse/golo/compiler/macro/MacroCache.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2012-2015 Institut National des Sciences Appliquées de Lyon (INSA-Lyon)
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.golo.compiler.macro;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.lang.invoke.MethodHandle;
+import java.util.function.Function;
+
+import org.eclipse.golo.compiler.ir.MacroInvocation;
+
+class MacroCache {
+  private final Map<MacroKey, MethodHandle> cache = new HashMap<>();
```

Comment:
Oh yes

> On 11 Jan 2016, at 14:21, Yannick Loiseau <notifications@github.com> wrote:
> 
> In src/main/java/org/eclipse/golo/compiler/macro/MacroCache.java:
> 
> > + * are made available under the terms of the Eclipse Public License v1.0
> > + * which accompanies this distribution, and is available at
> > + * http://www.eclipse.org/legal/epl-v10.html
> > + */
> > +
> > +package org.eclipse.golo.compiler.macro;
> > +
> > +import java.util.Map;
> > +import java.util.HashMap;
> > +import java.lang.invoke.MethodHandle;
> > +import java.util.function.Function;
> > +
> > +import org.eclipse.golo.compiler.ir.MacroInvocation;
> > +
> > +class MacroCache {
> > +  private final Map<MacroKey, MethodHandle> cache = new HashMap<>();
> 
> Since it's just a compile time cache, I'm not sure it worth the blow
> 
> —
> Reply to this email directly or view it on GitHub.
>